### PR TITLE
Do not change case of property name if inside a variable declaration in LESS

### PR DESCRIPTION
**Pull request by mvorisek**
## Description

Simple example:
```
@var: {
  camelCase: 0;
};

div {
  margin: @var[camelCase];
};
```

when `preserveCase` is changed to `preservecase`, LESS compilation fails.

This PR fixes such problem and handles also variable declaration with nested ruleset correctly.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Review by fisker — DISMISSED** [review]

**Comment by fisker** [comment]
@mvorisek The tests are failing.

**Comment by mvorisek** [comment]
> @mvorisek The tests are failing.

@fisker any idea how to fix it and why does it fail on Node 18 only?

**Comment by fisker** [comment]
`Use  `FULL_TEST` to debug https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md#deeper-testing

**Comment by mvorisek** [comment]
(my/added) test fixed, thanks for `FULL_TEST` hint

**Comment by fisker** [comment]
What cases the AST changes? I can't see what you changed.

**Comment by mvorisek** [comment]
Test to compile the LESS code from the PR description with `camelCase` property name changed to `camelcase`. LESS compilation will fail, case in a declaration must be preserved as it is case sensitive when referenced.

**Comment by fisker** [comment]
I how what are we fixing, I mean how did you fix the FULL_TEST, your code changes the AST at first, that's why the test failing. I want know how did you fix that.

**Comment by mvorisek** [comment]
sure, removed this not needed testcase I initially proposed:

image

**Comment by fisker** [comment]
I'm not familiar with LESS. Is it valid LESS code? Should we fix it instead of removing it.

**Comment by mvorisek** [comment]
here is minimalistic test case:

```
@varFollowedBySpace : {
  camelCase: 0;
};

div {
  margin: @varFollowedBySpace[camelCase];
};
```

in https://lesscss.org/less-preview/ I can compile it /w and /wo space to the same result

in https://prettier.io/playground/ when space is removed, the AST is different

not sure if really a bug in Prettier, maybe a bug in LESS parser? It is not related with this PR nor my needs, so feel free to pursue it, but I have no capacity to do so.

**Comment by fisker** [comment]
Can you add this test back, I can look into it later.

**Comment by fisker** [comment]
The difference is `.value` of `css-atrule`, right?

**Comment by fisker** [comment]
Thanks.

**Comment by mvorisek** [comment]
> The difference is `.value` of `css-atrule`, right?

I have added two failing test cases:

first is definitely in Prettier and should be easy to fix

2nd is the problem we discussed here, ie. different AST when at rule has a whitespace between colon:

```diff
-@var2 : {
+@var2: {
   notVar: 5;
 };
```

**Comment by fisker** [comment]
I'll take a look later, hopefully tomorrow.

File: tests/format/less/case/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -321,3 +321,131 @@ TABLE {
...
+: {
+  shouldnotchangecase: 5;
+}
+
+@not-var {
+  shouldnotchangecase: @var[ @notVarNested][notVar];
+  shouldnotchangecase: @var[ @notVarNested][notVar];
+  shouldnotchangecase: @var[ @notVarNested ][notVar];
+  shouldnotchangecase: @var[ @notVarNested][ notVar];
+  shouldnotchangecase: @var[ @notVarNested][notVar ];
```
Comment (fisker) [inline tests/format/less/case/__snapshots__/jsfmt.spec.js.snap:446]: We are fine with these spaces, right?

File: src/language-css/printer-postcss.js [hunk 2]
```diff
@@ -155,7 +155,12 @@ function genericPrint(path, options, print) {
 
       return [
         node.raws.before.replace(/[\s;]/g, ""),
-        insideICSSRuleNode(path) ? node.prop : maybeToLowerCase(node.prop),
+        (parentNode.type === "css-atrule" &&
+          typeof parentNode.raws.params === "string" &&
+          parentNode.raws.params.startsWith(":")) ||
```
Comment (fisker) [inline src/language-css/printer-postcss.js:160]: Should we use `parentNode.variable` instead?

Comment (fisker) [reply src/language-css/printer-postcss.js]: Ah, fixed cases with comments  https://github.com/prettier/prettier/pull/14034/commits/fd414fbd8dd79bc809f465c19374aec170e2b1c0

**Review by fisker — APPROVED** [review]

**Review by sosukesuzuki — APPROVED** [review]

**Comment by sosukesuzuki** [comment]
@mvorisek Are you okay with changes by @\fisker ? If you don't have concerns we can merge this PR.

File: src/language-css/printer-postcss.js (hunk 2 above)
Comment (mvorisek) [reply src/language-css/printer-postcss.js]: `parentNode.variable` seems fine/better

File: tests/format/less/case/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (mvorisek) [reply tests/format/less/case/__snapshots__/jsfmt.spec.js.snap]: I would expect no trailing space, especially when space is not present in the original. Can it be fixed easily?

**Comment by mvorisek** [comment]
https://github.com/prettier/prettier/pull/14034#discussion_r1059312395 would be nice to be fixed as well, otherwise I have no objections.

File: tests/format/less/case/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (fisker) [reply tests/format/less/case/__snapshots__/jsfmt.spec.js.snap]: I didn't notice the space is added.

**Comment by fisker** [comment]
https://github.com/prettier/prettier/pull/14034#discussion_r1059315108

Let's merge this first, we can try to fix it in other PR.

**Pull request merged**